gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientReadChannel.java

```diff
@@ -521,34 +571,70 @@ private void performPendingSeeks() {
       if (isInRangeSeek()) {
         skipInPlace();
       } else {
-        if (isRandomAccessPattern()) {
-          setRandomAccess();
-        }
         // close existing contentChannel as requested bytes can't be served from current
         // contentChannel;
         closeContentChannel();
       }
     }
 
+    private void updateAccessPattern() {
+      if (readOptions.getFadvise() == Fadvise.AUTO_RANDOM) {
+        if (isSequentialAccessPattern()) {
+          setSequentialAccess();
+        }
+      } else if (readOptions.getFadvise() == Fadvise.AUTO) {
+        if (isRandomAccessPattern()) {
+          setRandomAccess();
+        }
+      }
+    }
+
+    private boolean isSequentialAccessPattern() {
+      if (servedRequestLastIndex != -1) {
+        requestDistance.add(currentPosition - servedRequestLastIndex);
```

Comment:
This is bit of a code smell. A "get" method updating the state. Is there a way to avoid it?

Reply:
It's not updating the sate of content but it's internal state about how to access data. This is the feature we are offering with `AUTO` and `AUTO_RANDOM`.
Similarly, `read` operation updates the `currentPosition` pointer in file, that is also a state change.